File: swift/problem970.swift
class Solution {
    func powerfulIntegers(_ x: Int, _ y: Int, _ bound: Int) -> [Int] {
        let xpower = power(x,bound); let ypower = power(y,bound)
        var ans = Set<Int>()
        for xp in xpower {
            for yp in ypower {
                if (xp + yp <= bound) {
                    ans.insert(xp + yp)
                } else {
                    break
                }
            }
        }
        
        return Array(ans)
    }
    
    private func power(_ base: Int, _ bound: Int) -> [Int] {
        if (base == 1) {
            return [1]
        } else if (base == 0) {
            return [0]
        }
        var tmp = base; var ans = [1]
        
        while (tmp < bound) {
            ans.append(tmp)
            tmp *= base
        }
        
        return ans
    }
}

File: python/problem970.py
class Solution(object):
    def powerfulIntegers(self, x, y, bound):
        """
        :type x: int
        :type y: int
        :type bound: int
        :rtype: List[int]
        """
        sumxy = 0
        xi = 1
        
        li = set()
        while xi < bound:
            
            yi = 1
            while 1:
                
                sumxy = xi + yi
                if sumxy <= bound:
                    li.add(sumxy)
                else:
                    break
                if y == 1:
                    break
                else:
                    yi *= y
            if x == 1:
                break
            else:
                xi *= x
        return list(li)
